lib/rules/media-feature-name-no-unknown/index.js

```diff
@@ -52,129 +42,41 @@ const rule = (primary, secondaryOptions) => {
 			return;
 		}
 
-		/**
-		 * @param {import('postcss').AtRule} atRule
-		 * @param {string} featureName
-		 * @param {number} startIndex
-		 * @param {number} endIndex
-		 */
-		const validateFeatureName = (atRule, featureName, startIndex, endIndex) => {
-			if (!isStandardSyntaxMediaFeatureName(featureName) || isCustomMediaQuery(featureName)) {
-				return;
-			}
-
-			if (optionsMatches(secondaryOptions, 'ignoreMediaFeatureNames', featureName)) {
-				return;
-			}
-
-			if (vendor.prefix(featureName) || mediaFeatureNames.has(featureName.toLowerCase())) {
-				return;
-			}
-
-			const atRuleIndex = atRuleParamIndex(atRule);
-
-			report({
-				message: messages.rejected,
-				messageArgs: [featureName],
-				node: atRule,
-				index: atRuleIndex + startIndex,
-				endIndex: atRuleIndex + endIndex + 1,
-				ruleName,
-				result,
-			});
-		};
-
 		root.walkAtRules(/^media$/i, (atRule) => {
-			const mediaQueryList = parseMediaQuery(atRule);
-
-			mediaQueryList.forEach((mediaQuery) => {
-				if (isMediaQueryInvalid(mediaQuery)) return;
-
-				mediaQuery.walk(({ node }) => {
-					if (isMediaFeature(node)) {
-						const [, , startIndex, endIndex] = node.getNameToken();
-
-						validateFeatureName(atRule, node.getName(), startIndex, endIndex);
-
-						return;
-					}
-
-					if (isGeneralEnclosed(node)) {
-						const relevantTokens = topLevelTokenNodes(node);
-
-						if (!relevantTokens) {
-							return;
-						}
-
-						relevantTokens.forEach((token, i) => {
-							if (token[0] !== TokenType.Ident) {
-								return;
-							}
-
-							const nextToken = relevantTokens[i + 1];
-							const prevToken = relevantTokens[i - 1];
-
-							if (
-								// Media Feature
-								(!prevToken && nextToken && nextToken[0] === TokenType.Colon) ||
-								// Range Feature
-								(nextToken &&
-									nextToken[0] === TokenType.Delim &&
-									rangeFeatureOperator.test(nextToken[4].value)) ||
-								// Range Feature
-								(prevToken &&
-									prevToken[0] === TokenType.Delim &&
-									rangeFeatureOperator.test(prevToken[4].value))
-							) {
-								const [, , startIndex, endIndex, { value: featureName }] = token;
-
-								validateFeatureName(atRule, featureName, startIndex, endIndex);
-							}
-						});
-					}
+			const hasComments = atRule.raws.params && atRule.raws.params.raw;
+			const mediaRule = hasComments ? hasComments : atRule.params;
```

Comment:
[suggest] Modern operators can simplify the code like this:

```suggestion
			const mediaRule = atRule.raws.params?.raw ?? atRule.params;
```

Reply:
Ah, I forgot it, but we can use `getAtRuleParams()` utility instead: 😅 

https://github.com/stylelint/stylelint/blob/fd78ef7ad01123b0c4b05ad55d9986e3e1e21a01/lib/utils/getAtRuleParams.js#L7-L11

Reply:
We also can refactor the util, although it's out of the PR's scope.

```diff
 module.exports = function getAtRuleParams(atRule) {
-	const raws = atRule.raws;
-
-	return (raws.params && raws.params.raw) || atRule.params;
+	return atRule.raws.params?.raw ?? atRule.params;
 };
```

Reply:
Awesome!
This utility is exactly what was missing :)